PULUMI_FAILED_TESTS_DIR does not seem to work in pulumi-cloud

See: https://travis-ci.com/pulumi/pulumi-cloud/builds/64819755?utm_source=github_status&utm_medium=notification

It properly sets up the env variable:
```
$ export PULUMI_FAILED_TESTS_DIR=$(mktemp -d); echo "${PULUMI_FAILED_TESTS_DIR}"
/tmp/tmp.rMSvCle24m
```

Then the test fails during a integration test edit: 

```
[ mi/pulumi-cloud/aws/tests/unit ] **** Invoke '/opt/pulumi/bin/pulumi stack rm --yes p-it-travis-job-unit' in '/tmp/p-it-travis-job-unit-299479505'
[ mi/pulumi-cloud/aws/tests/unit ] Wrote output to /tmp/p-it-travis-job-unit-299479505/command-output/pulumi-stack-rm.20180201-090113.abad5.log
--- FAIL: Test_Examples (0.00s)
    --- FAIL: Test_Examples//home/travis/gopath/src/github.com/pulumi/pulumi-cloud/aws/tests/unit (2842.01s)
```

and the test states that it uploaded logs:

```
$ aws --region us-west-2 s3 cp --recursive "${PULUMI_FAILED_TESTS_DIR}" "s3://eng.pulumi.com/travis-logs/${TRAVIS_REPO_SLUG}/${TRAVIS_JOB_NUMBER}/failures"; echo "s3://eng.pulumi.com/travis-logs/${TRAVIS_REPO_SLUG}/${TRAVIS_JOB_NUMBER}/failures"
s3://eng.pulumi.com/travis-logs/pulumi/pulumi-cloud/1681.1/failures
after_failure.2
0.02s$ echo "one or more tests failed, to view detailed failure information, visit https://s3.console.aws.amazon.com/s3/buckets/eng.pulumi.com/travis-logs/${TRAVIS_REPO_SLUG}/${TRAVIS_JOB_NUMBER}/failures/"
one or more tests failed, to view detailed failure information, visit https://s3.console.aws.amazon.com/s3/buckets/eng.pulumi.com/travis-logs/pulumi/pulumi-cloud/1681.1/failures/
```

As there are no cp lines shown, it appears as if something has gone off hte rails here.

--- a/pkg/testing/integration/program.go
+++ b/pkg/testing/integration/program.go
@@ -399,7 +399,9 @@ func (pt *programTester) testLifeCycleInitAndDestroy() error {
 		return errors.Wrap(err, "running test preview, update, and edits")
 	}
 
-	keepTestDir = false
+	// Ran to completion. Delete the test directory if the test passed.
+	keepTestDir = pt.t.Failed()
+
 	return nil
 }
 

